### PR TITLE
Remove `SBF_JMP32` instruction class

### PR DESCRIPTION
**Pull request by LucasSte**
Although the instruction class `SBF_JMP32 = 0x06` exists in the upstream BPF target, it is absent in Solana's RBPF crate, and the code `0x06` has been repurposed for a new instruction class `PQR` in https://github.com/solana-labs/rbpf/pull/498.

This PR removes all instructions related to the class `SBF_JMP32`.

### CODE REVIEW

File: llvm/lib/Target/SBF/SBFISelLowering.cpp [hunk 1]
```diff
@@ -1091,23 +1089,20 @@ SBFTargetLowering::EmitInstrWithCustomInserter(MachineInstr &MI,
...
-  SET_NEWCC(SETUGT, JUGT);
-  SET_NEWCC(SETGE, JSGE);
-  SET_NEWCC(SETUGE, JUGE);
-  SET_NEWCC(SETEQ, JEQ);
-  SET_NEWCC(SETNE, JNE);
-  SET_NEWCC(SETLT, JSLT);
-  SET_NEWCC(SETULT, JULT);
-  SET_NEWCC(SETLE, JSLE);
-  SET_NEWCC(SETULE, JULE);
+    SET_NEWCC(SETGT, JSGT);
```
Comment (dmakarov) [inline llvm/lib/Target/SBF/SBFISelLowering.cpp]: I think this reindentation is wrong. The macro invocations should be indented to the same level as `case` and `default` labels.

File: llvm/test/CodeGen/SBF/atomics_sbf.ll [hunk 2]
```diff
@@ -181,8 +181,14 @@ entry:
...
 ; CHECK: ldxw w0, [r1 + 0]
-; CHECK: mov32 w3, w0
-; CHECK: jslt w0, w2,
+; CHECK: mov64 r4, r0
+; CHECK: lsh64 r4, 32
+; CHECK: arsh64 r4, 32
+; CHECK: mov32 r5, w2
+; CHECK: lsh64 r5, 32
+; CHECK: arsh64 r5, 32
+; CHECK:mov32 w3, w0
```
Comment (dmakarov) [inline llvm/test/CodeGen/SBF/atomics_sbf.ll]: shouldn't here be a space between `CHECK:` and `mov32`?

**Review by dmakarov — COMMENTED** [review]
LGTM, except a couple of formatting nits.

File: llvm/lib/Target/SBF/SBFISelLowering.cpp (hunk 1 above)
Comment (LucasSte) [reply llvm/lib/Target/SBF/SBFISelLowering.cpp]: I got this indentation from `clang-format` and thought it was right. I'll revert to the original then.

File: llvm/test/CodeGen/SBF/atomics_sbf.ll (hunk 2 above)
Comment (LucasSte) [reply llvm/test/CodeGen/SBF/atomics_sbf.ll]: That slipped through.

**Review by dmakarov — APPROVED** [review]

**Pull request merged**